apps/pie-docs/src/_11ty/shortcodes/card.js

```diff
@@ -16,30 +20,101 @@ module.exports = function ({ items }) {
         },
     });
 
+    return isInternalLink ? internalLinkIcon : externalLinkIcon;
+};
+
+const buildCardLabel = (linkText, cardHasImage, href, shouldOpenInNewTab, isInternalLink) => {
+    const labelClasses = [
+        'c-card-labelContainer',
+        cardHasImage && 'c-card-labelContainer--hasImage',
+        isInternalLink && 'c-card-labelContainer--internalLink'
+    ].filter(Boolean).join(' ');
+
+    const target = shouldOpenInNewTab ? 'target="_blank"' : '';
+
+    return `<div class="${labelClasses}">
+                ${href
+        ? `<a class="c-card-label" href="${href}" ${target}><span>${linkText}</span></a>
+                        ${buildLinkIcon(isInternalLink)}`
+        : `<p class="c-card-label">${linkText}</p>`
```

Comment:
even if its not a link we probably still need to truncate, which is contingent on that span in the anchor element above